trunk/SUAVE/Optimization/Package_Setups/differential_evolution.py

```diff
@@ -0,0 +1,1362 @@
+## @ingroup Optimization-Package_Setups
```

Comment:
why are you copying their code?

If you need a new class based on theirs import it and modify/add the functions that need so

This will fix your coveralls issue

Reply:
They recently added constraints but it was not in the same format we use in SUAVE. To allow the constraints to be passed as a data structure that can be evaluated I had to modify the majority of the functions in the SciPy Differential Evolution Script. If you did a compare them with diff_checker(or any online file checker you will see the changes). In a nutshell, the DE code needs to be able to recursively evaluate constraints and store in a particular format.

Reply:
After looking through this, I think there is a better way.

I don't like the idea of adding a diverged chunk of code from a different package into SUAVE. This could be a real headache for us.

It looks like the differential evolution just needs a tuple of nonlinear constraints passed that behave as functions. I think that we can just just dynamically generate constraints that map to the nexus with just an extra function added to the scipy setup